modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/EditAPIApplication.tsx

```diff
@@ -47,7 +47,7 @@ export default function EditAPIApplication({
 	const currentAPIApplicationID = getCurrentURLParamValue({
 		paramSufix: 'apiApplicationId',
 		portletId,
-	});
+	}) as string;
```

Comment:
Instead of forcing the type (which could be null) can we throw an exception or fail somehow?

I know this scenario is almost impossible to happen, but tricking the type system may not be the best idea.

Reply:
While very unlikely, you can just accept that the value can be null. Delete the "as string" above and...
```
interface APIApplicationsTableProps {
	apiApplicationBaseURL: string;
	apiURLPaths: APIURLPaths;
	currentApplicationId: string | null;
	portletId: string;
	readOnly: boolean;
}
```
If it comes null, than the query will be `"r_apiApplicationToAPIEndpoints_c_apiApplicationId eq 'null'"` which will return no items (right?) and that's ok. If it happens, this would be a bug in how we'are adding the ID to the URL, not in the fix for the bug we are discussing here.

Reply:
Makes sense.

Reply:
Make sense @cgmonte!!